USB: netusb: Use standard frame size and cleanup

finikorg opened the pull request:
Use standard frame size, correct buffer sizes and cleanup.

Fixes #14780

finikorg committed:
usb: netusb: Cleanup USB Ethernet networking

Cleanup ECM, EEM and RNDIS networking functions, removing unneeded
headers and unused definitions.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: rndis: Add check for large packets

Drop packets larger then tx_buf.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: ecm: Use NET_ETH_MAX_FRAME_SIZE as frame size

Use standard defined frame size for buffer and wMaxSegmentSize
parameter. Changes frame size from 1522 to 1514.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: ecm: Add check for large packets

Make sure we do not overwrite tx_buf.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: eem: Use NET_ETH_MAX_FRAME_SIZE as frame size

Use standard NET_ETH_MAX_FRAME_SIZE for frame size calculation. This
changes buffer from 1522 to 1514 + 2 + 4. Remove also magic numbers.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: eem: Add check for large packets

Make sure we do not overwrite tx_buf.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: rndis: Fix maximum RNDIS frame size

Use standard NET_ETH_MAX_FRAME_SIZE in calculation adding space for
ethernet header.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

jfischer-phytec-iot commented on subsys/usb/class/netusb/function_rndis.c:
```diff
@@ -884,7 +867,7 @@ static int handle_encapsulated_cmd(u8_t *data, u32_t len)
 }
 
-#if SEND_MEDIA_STATUS
+#if IS_ENABLED(SEND_MEDIA_STATUS)
```
Could you please add a comment why is it disable per default?

jukkar replied:
I think this should be `#if defined(SEND_MEDIA_STATUS)` instead.

finikorg replied:
> I think this should be `#if defined(SEND_MEDIA_STATUS)` instead.
I have `SEND_MEDIA_STATUS` defined to `0`.

finikorg replied:
> Could you please add a comment why is it disabled per default?

For some reason this option does not work with Windows (8), probably I miss something but it works fine without this media status, probably we do not need it. Maybe I just remove it completely.

finikorg commented:
@ubicore Please check this modifications.

finikorg committed:
usb: netusb: rndis: Remove old unused code

Remove old unused code sending media status, sending this packet
causes Windows problems although Linux works with it. Removing this
code does not affect RNDIS functionality.

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>

finikorg committed:
usb: netusb: ecm: Print packet hexdump only for VERBOSE_DEBUG

Use VERBOSE_DEBUG to enable net_pkt_hexdump().

Signed-off-by: Andrei Emeltchenko <andrei.emeltchenko@intel.com>
